nodebuilder/pruner/module.go

```diff
@@ -36,9 +36,13 @@ func ConstructModule(tp node.Type, cfg *Config) fx.Option {
 	)
 
 	baseComponents := fx.Options(
-		fx.Supply(cfg),
+		// supply the default config, which can only be overridden by
+		// passing the `--archival` flag
+		fx.Supply(&Config{
+			EnableService: true,
+		}),
```

Comment:
nit: Default config would look better as `func DefaultConfig() config` and would not need comments